ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/network/DefaultP2PNetwork.java

```diff
@@ -778,7 +761,6 @@ private void validate() {
       checkState(
           supportedCapabilities != null && supportedCapabilities.size() > 0,
           "Supported capabilities must be set and non-empty.");
-      checkState(peerBlacklist != null, "PeerBlacklist must be set.");
```

Comment:
do we handle null peerPermissions correctly?

Reply:
I think so - `peerPermissions` gets initialized with a default value, and the setter does a null check so that it can't get overwritten with a null value.